File: CivitasRuby/lib/sorpresa_pagar_cobrar.rb
#encoding:utf-8

require_relative 'jugador'
require_relative 'tablero'
require_relative 'casilla'
require_relative 'mazo_sorpresas'
require_relative 'sorpresa'
require 'sorpresa.rb'

module Civitas
    class SorpresaPagarCobrar < Sorpresa
        def initialize(valor,texto)
            super(texto)
            @valor=valor
        end

        def aplicar_a_jugador(actual, todos)
            if(jugador_correcto(actual, todos))
                informe(actual, todos)
                todos[actual].modifica_saldo(@valor)
            end
        end

        public_class_method :new
    end
end

File: CivitasJava/src/civitas/SorpresaPagarCobrar.java
package civitas;

import java.util.ArrayList;

public class SorpresaPagarCobrar extends Sorpresa{
 
    SorpresaPagarCobrar(int valor, String texto){
        super(texto);
        this.valor=valor;
    }

    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            todos.get(actual).modificarSaldo(valor);
        }
    }
}